### PR TITLE
Use query builder instead of OC_DB in user_ldap 

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref #23929

### CODE REVIEW

**Review by blizzz — CHANGES_REQUESTED** [review]
there's an array to string conversion issue on deleting configs

File: apps/user_ldap/lib/Helper.php [hunk 1]
```diff
@@ -160,30 +162,22 @@ public function deleteServerConfiguration($prefix) {
...
-		}
+		$query = $this->connection->getQueryBuilder();
+		$query->delete('appconfig')
+			->where($query->expr()->eq('appid', $query->createNamedParameter('user_ldap')))
+			->andWhere($query->expr()->notIn('configkey', $query->createNamedParameter([
+				'enabled',
+				'installed_version',
+				'types',
+				'bgjUpdateGroupsLastRun',
+			]), IQueryBuilder::PARAM_STR_ARRAY));
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Helper.php:173]:
```suggestion
			->andWhere($query->expr()->notIn('configkey', $query->createNamedParameter([
				'enabled',
				'installed_version',
				'types',
				'bgjUpdateGroupsLastRun',
			], IQueryBuilder::PARAM_STR_ARRAY)));
```

**Review by ChristophWurst — APPROVED** [review]

**Review by blizzz — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Mind to add the DB group to the tests in `apps/user_ldap/tests/HelperTest.php` as the DB access is now detected and blocked ... so they were actually executed in the wrong unit test before 🙈

**Comment by skjnldsv** [comment]
Lots of ldap issues in the NODB tests




```
1) OCA\User_LDAP\Tests\HelperTest::testGetServerConfigurationPrefixes
277 | Your test case is not allowed to access the database.
278 |  
279 | /drone/src/tests/lib/TestCase.php:133
280 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:155
281 | /drone/src/3rdparty/pimple/pimple/src/Pimple/Container.php:118
282 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:122
283 | /drone/src/lib/private/ServerContainer.php:156
284 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:56
285 | /drone/src/lib/private/Server.php:1721
286 | /drone/src/apps/user_ldap/tests/HelperTest.php:43
287 |  
288 | 2) OCA\User_LDAP\Tests\HelperTest::testGetServerConfigurationPrefixesActive
289 | Your test case is not allowed to access the database.
290 |  
291 | /drone/src/tests/lib/TestCase.php:133
292 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:155
293 | /drone/src/3rdparty/pimple/pimple/src/Pimple/Container.php:118
294 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:122
295 | /drone/src/lib/private/ServerContainer.php:156
296 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:56
297 | /drone/src/lib/private/Server.php:1721
298 | /drone/src/apps/user_ldap/tests/HelperTest.php:43
299 |  
300 | 3) OCA\User_LDAP\Tests\HelperTest::testGetServerConfigurationHost
301 | Your test case is not allowed to access the database.
302 |  
303 | /drone/src/tests/lib/TestCase.php:133
304 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:155
305 | /drone/src/3rdparty/pimple/pimple/src/Pimple/Container.php:118
306 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:122
307 | /drone/src/lib/private/ServerContainer.php:156
308 | /drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:56
309 | /drone/src/lib/private/Server.php:1721
310 | /drone/src/apps/user_ldap/tests/HelperTest.php:43
311


```

**Review by MorrisJobke — APPROVED** [review]
Tested LDAP wizard and provisioning and all still works 👍

**Pull request merged**